internal/grpcsync/callback_serializer.go

```diff
@@ -56,8 +56,12 @@ func (t *CallbackSerializer) run(ctx context.Context) {
 	for ctx.Err() == nil {
 		select {
 		case <-ctx.Done():
+			t.callbacks.Close()
 			return
-		case callback := <-t.callbacks.Get():
+		case callback, ok := <-t.callbacks.Get():
+			if !ok {
+				return
+			}
```

Comment:
Similar comment to a previous one. Is handling the second return value from the channel receive necessary here? We close the `callbacks` buffer in this same function, and when we do that, we return from this function.